### PR TITLE
Fix overcounting error when exploding ad fields

### PR DESCRIPTION
**Pull request by benmiroglio**
I found an error when looking at the ad probes using search clients daily versus main_summary. There is an over-counting error when exploding those fields.

The error/fix can be more easily explored in this minimal reproduction:

https://dbc-caf9527b-e073.cloud.databricks.com/#notebook/141261/command/141311

@harterrt r?

### CODE REVIEW

**Comment by benmiroglio** [comment]
also tagging @wlach for review since you've worked on this dataset recently.

File: mozetl/search/aggregates.py [hunk 1]
```diff
@@ -221,10 +221,16 @@ def _drop_source_columns(base):
             derived = derived.drop(source_col)
         return derived
 
-    def _select_counts(main_summary, col_name, count_udf=None):
-        derived = main_summary.withColumn(
-            "single_search_count", explode(col("search_counts"))
-        ).filter("single_search_count.count < %s" % MAX_CLIENT_SEARCH_COUNT)
+    def _select_counts(
+        main_summary, col_name, count_udf=None, get_single_search_count=True
```
[the remark this answers is not in the document]
Comment (benmiroglio) [reply mozetl/search/aggregates.py]: Certainly not, it was just the minimal amount of code needed to fix the issue. It probably makes sense to break the aggregations out for clarity.

**Comment by benmiroglio** [comment]
@wlach I removed the parameter and used the column name to apply the same logic. Breaking this out into two different functions seemed like overkill after prototyping. 

Regarding adding tests, I took a look at the tests and it became clear it will require a good amount of cognitive load for me to figure out how to properly account for this edge case. I can get around to it eventually but I'll defer to you on whether we should block this PR until the test properly account for this.

File: mozetl/search/aggregates.py [hunk 2]
```diff
@@ -222,15 +222,20 @@ def _drop_source_columns(base):
...
-            "single_search_count", explode(col("search_counts"))
-        ).filter("single_search_count.count < %s" % MAX_CLIENT_SEARCH_COUNT)
+        ad_fields = ("ad_click", "search_with_ads")
+        if col_name in ad_fields:
+            # special case for ad click probes
+            derived = main_summary
+        else:
+            derived = main_summary.withColumn(
+                "single_search_count", explode(col("search_counts"))
+            ).filter("single_search_count.count < %s" % MAX_CLIENT_SEARCH_COUNT)
```
Comment (wlach) [inline mozetl/search/aggregates.py]: I don't think we really need to specify the ad click fields. Really it's the single search count case which is "special". I think I would prefer something like this:

```py
if col_name == "single_search_count":
    derived = main_summary.withColumn(
                "single_search_count", explode(col("search_counts"))
            ).filter("single_search_count.count < %s" % MAX_CLIENT_SEARCH_COUNT)
else:
    derived = main_summary
```

File: mozetl/search/aggregates.py [hunk 3]
```diff
@@ -222,15 +222,20 @@ def _drop_source_columns(base):
...
         if count_udf is not None:
             derived = derived.withColumn(col_name, explode(count_udf))
         derived = _drop_source_columns(
             derived.select(["*"] + _get_search_fields(col_name)).drop(col_name)
         )
-        if col_name is not "single_search_count":
+        if col_name is not "single_search_count" or col_name in ad_fields:
```
Comment (wlach) [inline mozetl/search/aggregates.py]: I don't think this extra logic is necessary. If the col_name is one of the ad fields, by definition it is not "single_search_count".

Comment (benmiroglio) [reply mozetl/search/aggregates.py]: True, this will just drop a phantom column in the case of the ad_clicks, but I was being (perhaps too) explicit.

File: mozetl/search/aggregates.py (hunk 2 above)
Comment (benmiroglio) [reply mozetl/search/aggregates.py]: You're right, this is cleaner!

File: mozetl/search/aggregates.py (hunk 3 above)
Comment (benmiroglio) [reply mozetl/search/aggregates.py]: Actually, I think this if-statement can be omitted altogether with the added logic at the top of the function.

**Comment by wlach** [comment]
This looks good! Thank you Ben. We should probably backfill the data into the past to make sure we're using valid data here, I'll file a bug about that tomorrow. For now, I'll merge this.
